modules/openapi-generator/src/main/resources/typescript/api/api.mustache

```diff
@@ -59,7 +59,11 @@ export class {{classname}}RequestFactory extends BaseAPIRequestFactory {
             .replace('{' + '{{baseName}}' + '}', encodeURIComponent(String({{paramName}}))){{/pathParams}};
 
         // Make Request Context
-        const requestContext = _config.baseServer.makeRequestContext(localVarPath, HttpMethod.{{httpMethod}});
+        const requestContext = _config.baseServer.makeRequestContext(
+            localVarPath,
+            HttpMethod.{{httpMethod}},
+            _config.baseServer.getHeaders()
```

Comment:
looks like you always pass baseServer.headers to the baseServer.makeRequestContext call.
looks redundant - why don't baseServer.makeRequestContext does it internally - automatically?